libraries/bot-connector/src/test/java/com/microsoft/bot/connector/authentication/ParameterizedBotFrameworkAuthenticationTest.java

```diff
@@ -0,0 +1,205 @@
+package com.microsoft.bot.connector.authentication;
+
+import com.microsoft.bot.connector.Channels;
+import com.microsoft.bot.schema.Activity;
+
+import com.microsoft.bot.schema.ChannelAccount;
+import com.microsoft.bot.schema.RoleTypes;
+import okhttp3.OkHttpClient;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+
+import static org.junit.Assert.*;
+
+public class ParameterizedBotFrameworkAuthenticationTest {
+    @Test
+    public void getOriginatingAudience_shouldBeCorrectlyFilled() {
+        PasswordServiceClientCredentialFactory passwordServiceClientCredentialFactory = new PasswordServiceClientCredentialFactory("42",
+            "test");
+
+        AuthenticationConfiguration authenticationConfiguration = new AuthenticationConfiguration();
+        assertEquals("test@example.org",
+            (new ParameterizedBotFrameworkAuthentication(true, "test@example.org", "test@example.org", "ABC123",
+                "https://example.org/example", "test@example.org", "test@example.org", "42", passwordServiceClientCredentialFactory,
+                authenticationConfiguration, new OkHttpClient())).getOriginatingAudience());
+    }
+
+    @Test
+    public void authenticateChannelRequest_withAuthenticationEnabled_shouldNotBeNull() {
+        PasswordServiceClientCredentialFactory passwordServiceClientCredentialFactory = new PasswordServiceClientCredentialFactory("42",
+            "test");
+
+        AuthenticationConfiguration authenticationConfiguration = new AuthenticationConfiguration();
+        ParameterizedBotFrameworkAuthentication parameterizedBotFrameworkAuthentication = new ParameterizedBotFrameworkAuthentication(true, "test@example.org", "test@example.org", "ABC123",
+            "https://example.org/example", "test@example.org", "test@example.org", "42", passwordServiceClientCredentialFactory,
+            authenticationConfiguration, new OkHttpClient());
+
+        ClaimsIdentity mockClaimsIdentity = Mockito.mock(ClaimsIdentity.class);
+        ParameterizedBotFrameworkAuthentication spyParameterizedBotFrameworkAuthentication = Mockito.spy(parameterizedBotFrameworkAuthentication);
+        Mockito.doReturn(CompletableFuture.completedFuture(mockClaimsIdentity))
+            .when(spyParameterizedBotFrameworkAuthentication)
+            .jwtTokenValidationValidateAuthHeader("Auth Header", "unknown", null);
+        ClaimsIdentity claims = spyParameterizedBotFrameworkAuthentication
+            .authenticateChannelRequest("Auth Header").join();
+
+        assertNotNull(claims);
```

Comment:
Can any other assertion be added to validate that aside from not being `NULL`, `claims `contains the information that we expect to have?